include/up-cpp/transport/UTransport.h

```diff
@@ -0,0 +1,94 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * 
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _UTRANSPORT_
+#define _UTRANSPORT_
+
+#include <stdint.h>
+#include <cstddef>
+#include <up-cpp/transport/UListener.h>
+#include <up-cpp/transport/datamodel/UPayload.h>
+#include <up-core-api/uri.pb.h>
+#include <up-core-api/ustatus.pb.h>
+#include <up-core-api/uattributes.pb.h>
+
+namespace uprotocol::utransport {
+
+    class UTransport {
+
+        public:
+
+            /**
+             * Transmit UPayload to the topic using the attributes defined in
+             * UTransportAttributes.
+             *
+             * @param message message to be sent
+             *
+             * @return Returns OKSTATUS if the payload has been successfully 
+             *         sent (ACK'ed), otherwise it returns FAILSTATUS with the
+             *         appropriate failure.
+             */
+            virtual uprotocol::v1::UStatus send(
+                    const uprotocol::utransport::UMessage &message) = 0;
+
+            /**
+             * Register listener to be called when UPayload is received for the
+             * specific topic.
+             *
+             * @param topic Resolved UUri for where the message arrived via the
+             *              underlying transport technology.
+             * @param listener The method to execute to process the date for
+             *                 the topic.
+             *
+             * @return Returns OKSTATUS if the listener is unregistered
+             *         correctly, otherwise it returns FAILSTATUS
+             *         with the appropriate failure.
+             */      
+            virtual uprotocol::v1::UStatus registerListener(
+                    const uprotocol::v1::UUri &uri,
+                    const uprotocol::utransport::UListener &listener) = 0;
+
+            /**
+             * Unregister a listener for a given topic. Messages arriving on
+             * this topic will no longer be processed by this listener.
+             *
+             * @param topic Resolved UUri for where the listener was registered
+             *              to receive messages from.
+             * @param listener The method to execute to process the date for
+             *                 the topic.
+             *
+             * @return Returns OKSTATUS if the listener is unregistered
+             *         correctly, otherwise it returns FAILSTATUS with the
+             *         appropriate failure.
+             */
+            virtual uprotocol::v1::UStatus unregisterListener(
+                    const uprotocol::v1::UUri &uri, 
+                    const uprotocol::utransport::UListener &listener) = 0;
+
```

Comment:
According to the spec, there should be a Receive() method that isn't implemented at all.